File: HomeMaintenanceSchedule/src/HomeMaintenanceSchedule.Core/Model/TaskAggregate/Enums/MaintenanceType.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace HomeMaintenanceSchedule.Core;

/// <summary>
/// Represents the type of maintenance.
/// </summary>
public enum MaintenanceType
{
    /// <summary>
    /// Routine preventive maintenance.
    /// </summary>
    Preventive = 0,

    /// <summary>
    /// Corrective or repair maintenance.
    /// </summary>
    Corrective = 1,

    /// <summary>
    /// Seasonal maintenance.
    /// </summary>
    Seasonal = 2,

    /// <summary>
    /// Emergency maintenance.
    /// </summary>
    Emergency = 3,

    /// <summary>
    /// Inspection or assessment.
    /// </summary>
    Inspection = 4,
}

File: HomeMaintenanceSchedule/src/HomeMaintenanceSchedule.WebApp/projects/home-maintenance-schedule/src/app/models/maintenance-type.ts
export enum MaintenanceType {
  Preventive = 0,
  Corrective = 1,
  Seasonal = 2,
  Emergency = 3,
  Inspection = 4
}

export const MAINTENANCE_TYPE_LABELS: Record<MaintenanceType, string> = {
  [MaintenanceType.Preventive]: 'Preventive',
  [MaintenanceType.Corrective]: 'Corrective',
  [MaintenanceType.Seasonal]: 'Seasonal',
  [MaintenanceType.Emergency]: 'Emergency',
  [MaintenanceType.Inspection]: 'Inspection'
};
